pkg/deploy/chart_manager.go

```diff
@@ -155,6 +156,10 @@ func (m *EnabledSKRWebhookChartManager) resolveKcpAddr(ctx context.Context, kcpC
 	if m.kcpAddr != "" {
 		return m.kcpAddr, nil
 	}
+	if m.config.WatcherLocalTestingEnabled {
```

Comment:
I think you should explain the flag here a bit more. Is there no other way we can test this setup without the flag? I would really like to avoid testing flags in the main binary if possible.

Reply:
I agree that we should avoid using flags if possible. that's why I have thought about discovering if we are running in the local-testing-mode or not. But I faced the issue that k3d assigns an external IP to the Istio GW which doesn't work with the watcher when it tries to send KCP requests.

Reply:
So the thing is the Istio GW only assigns that IP if you use a service of type load balancer. If you change this to a different SVC type it should not assign any IP afaik.

Reply:
Related to the below discussion since you mentioned you dont want to use NodePort, so maybe lets continue that there.